src/js/prop-types/locale-prop-type.js

```diff
@@ -1,7 +1,6 @@
 import isRequiredHelper from './is-required-helper';
 
-// link https://stackoverflow.com/questions/8758340/regex-to-detect-locales
-const reLocale = /^[a-z]{2,4}(-([A-Z][a-z]{3}|[0-9]{3}))?(-([A-Z]{2}|[0-9]{3}))?$/;
+const reLocale = /^([\w]{2})-([\w]{2})/;
```

Comment:
does this validate if the last two letters are uppercase?

Reply:
good check! I just tested it and seems like it works:

<img width="354" alt="bildschirmfoto 2018-12-06 um 12 03 46" src="https://user-images.githubusercontent.com/275235/49580379-1384c100-f94f-11e8-9d17-363cd27fe7f6.png">

Reply:
hmh, it does also accept lower case country codes `de-ch`.
https://regex101.com/r/UfQQOu/1

<img width="870" alt="screen shot 2018-12-06 at 12 10 30" src="https://user-images.githubusercontent.com/914443/49580688-cce39680-f94f-11e8-9ca3-f2d87ed14b23.png">

Reply:
@AndyOGo yepp. Otherwise it crashes. The intl api accepts both so i guess is ok

Reply:
@AndyOGo mmm im actually worried now. On chrome it accepts `de-de` and `de-DE`. I was overviewving the spec https://www.ecma-international.org/ecma-402/1.0/#sec-6.2 and couldnt get clever about what is allowed and what not

Reply:
Yepp ok so the regex is good as is as the specs contradict iteslf and at the end if intl api can work with it, lets set it as intl api want it

Reply:
hehe, yeah. Many people/organisations have tried to solve i18n problems and spec them, but these are very complex and changing over time

Reply:
And a very well distributed and accepted engeeiring principle says:

Flexible im entegennehmen und präxis im weitergeben